metals/src/main/scala/scala/meta/internal/builds/ShellRunner.scala

```diff
@@ -130,3 +133,80 @@ class ShellRunner(
   }
 
 }
+
+object ShellRunner {
+
+  /**
+   * Note: if returning a [[Future]] is more suitable, all you
+   * need to do is to uncomment the commented code.
+   */
+  def quickRunWithShortStringResult(
+      //        commandRun: String,
+      args: List[String],
+      directory: AbsolutePath,
+      redirectErrorOutput: Boolean,
+      additionalEnv: Map[String, String] = Map.empty,
+      processOut: String => Unit = scribe.info(_),
+      processErr: String => Unit = scribe.error(_),
+      propagateError: Boolean = false,
+      maybeJavaHome: Option[String]
+
+      //       logInfo: Boolean = true
+  ):
+  // Future[
+  Option[String]
+  //  ]
+  = {
+    //   val elapsed = new Timer(time)
+
+    val env = additionalEnv ++ maybeJavaHome.map("JAVA_HOME" -> _).toMap
+    val ps = SystemProcess.run(
+      args,
+      directory,
+      redirectErrorOutput,
+      env,
+      Some(processOut),
+      Some(processErr),
+      propagateError
+    )
+    //    // NOTE(olafur): older versions of VS Code don't respect cancellation of
+    //    // window/showMessageRequest, meaning the "cancel build import" button
+    //    // stays forever in view even after successful build import. In newer
+    //    // VS Code versions the message is hidden after a delay.
+    //    val taskResponse =
+    //      languageClient.metalsSlowTask(
+    //        new MetalsSlowTaskParams(commandRun)
+    //      )
+
+    //    val result = Promise[Option[String]]
+    //    taskResponse.asScala.foreach { item =>
+    //      if (item.cancel) {
+    //        if (logInfo)
+    //          scribe.info(s"user cancelled $commandRun")
+    //        result.trySuccess(None)
+    //        ps.cancel
+    //      }
+    //    }
+
+    //   val processFuture = ps.complete
+
+    //    cancelables
+    //      .add(() => ps.cancel)
+    //      .add(() => taskResponse.cancel(false))
+
+    //    processFuture.map { code =>
+    //      taskResponse.cancel(false)
+    //      if (logInfo)
+    //        scribe.info(s"time: ran '$commandRun' in $elapsed")
+    //     result.trySuccess(code match {
+    //       case ExitCodes.Success =>
+
+    val scanner = new Scanner(ps.inputStream).useDelimiter("\\A")
```

Comment:
Could you maybe move the reading of the string to `JdkVersion` and test it with unit tests?

I don't think it will work with this sample output:
```
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment Temurin-17.0.2+8 (build 17.0.2+8)
OpenJDK 64-Bit Server VM Temurin-17.0.2+8 (build 17.0.2+8, mixed mode, sharing)
```
did you test it?

Reply:
You are right. from pprint:

> the shell runner obtained java version as OpenJDK 64-Bit Server VM Homebrew (build 18.0.1.1+0, mixed mode, sharing)

Reply:
Hey, instead of going the `ShellRunner` path, do you think it would have been a better idea to simply do `System.getProperty("java.version”)` according to [this](https://stackoverflow.com/questions/5103121/how-to-find-the-jvm-version-from-a-program) ?

In my case, it is produces just  `18.0.1.1`

which does not require any extraction :)